util/test/tests/Vulkan/VK_Dual_Source.py

```diff
@@ -0,0 +1,128 @@
+    demos_test_name = 'VK_Dual_Source'
+    demos_frame_cap = 5
+
+    def check_capture(self):
+        apiprops: rd.APIProperties = self.controller.GetAPIProperties()
+
+        if not apiprops.pixelHistory:
+            rdtest.log.print("Vulkan pixel history not tested")
+            return
+
+        self.primary_test()
+
+    def primary_test(self):
+        test_marker: rd.ActionDescription = self.find_action("End test B2")
+        self.controller.SetFrameEvent(test_marker.eventId, True)
+
+        pipe: rd.PipeState = self.controller.GetPipelineState()
+
+        rt: rd.BoundResource = pipe.GetOutputTargets()[0]
+
+        tex = rt.resourceId
+        tex_details = self.get_texture(tex)
+
+        sub = rd.Subresource()
+        if tex_details.arraysize > 1:
+            sub.slice = rt.firstSlice
+        if tex_details.mips > 1:
+            sub.mip = rt.firstMip
+
+        # Pixel inside first triangle
+        modifs_1: List[rd.PixelModification] = self.controller.PixelHistory(tex, 175, 125, sub, rt.typeCast)
+        # Pixel inside second triangle
+        modifs_2: List[rd.PixelModification] = self.controller.PixelHistory(tex, 275, 175, sub, rt.typeCast)
+        # Pixel inside both triangles
+        modifs_both: List[rd.PixelModification] = self.controller.PixelHistory(tex, 225, 150, sub, rt.typeCast)
+
+        self.check_range(modifs_1, modifs_2, modifs_both, "A1")
+        self.check_range(modifs_1, modifs_2, modifs_both, "A2")
+        self.check_range(modifs_1, modifs_2, modifs_both, "B1")
+        self.check_range(modifs_1, modifs_2, modifs_both, "B2")
+
+    def check_range(self, orig_modifs_1, orig_modifs_2, orig_modifs_both, group):
+        start_eid = self.find_action("Begin test " + group).eventId
+        end_eid = self.find_action("End test " + group).eventId
+        modifs_1 = [m for m in orig_modifs_1 if m.eventId > start_eid and m.eventId < end_eid]
+        modifs_2 = [m for m in orig_modifs_2 if m.eventId > start_eid and m.eventId < end_eid]
+        modifs_both = [m for m in orig_modifs_both if m.eventId > start_eid and m.eventId < end_eid]
+        if not len(modifs_1) == 1:
+            raise rdtest.TestFailureException(
+                "group {}: expected 1 modification but found {}".format(group, len(modifs_1)))
+        if not len(modifs_2) == 1:
+            raise rdtest.TestFailureException(
+                "group {}: expected 1 modification but found {}".format(group, len(modifs_2)))
+        if not len(modifs_both) == 2:
+            raise rdtest.TestFailureException(
+                "group {}: expected 2 modifications but found {}".format(group, len(modifs_both)))
+        modif_1 = modifs_1[0]
+        modif_2 = modifs_2[0]
+        modif_both_1 = modifs_both[0]
+        modif_both_2 = modifs_both[1]
+        # modif_both_1 should match modif_1 as both are the same triangle on the same background
+        assert(value_selector(modif_1.preMod.col) == value_selector(modif_both_1.preMod.col))
```

Comment:
I'm using `assert` here because it seems like it would be a lot easier to use it than to do a check and then throw a `rdtest.TestFailureException` with a meaningful message. If needed I can change it, but I fear it would make things more confusing.